src/overlays/actors/ovl_En_Butte/z_en_butte.c

```diff
@@ -21,41 +45,376 @@ const ActorInit En_Butte_InitVars = {
+    /* 0x04 */ f32 speedXZTarget;
+    /* 0x08 */ f32 speedXZScale;
+    /* 0x0C */ f32 speedXZStep;
+    /* 0x10 */ s16 rotYStep;
+} EnButteFlightParams; // size = 0x14
+
+static EnButteFlightParams sFlyAroundParams[] = {
+    { 5, 35, 0.0f, 0.1f, 0.5f, 0 },
+    { 10, 45, 1.1f, 0.1f, 0.25f, 1000 },
+    { 10, 40, 1.5f, 0.1f, 0.3f, 2000 },
+};
+static EnButteFlightParams sFollowLinkParams[] = {
+    { 3, 3, 0.8f, 0.1f, 0.2f, 0 },
+    { 10, 20, 2.0f, 0.3f, 1.0f, 0 },
+    { 10, 20, 2.4f, 0.3f, 1.0f, 0 },
+};
+
+extern AnimationHeader D_05002470;
+extern SkeletonHeader D_050036F0;
+
+void EnButte_SelectFlightParams(EnButte* this, EnButteFlightParams* flightParams) {
+    if (this->flightParamsIdx == 0) {
+        if (Math_Rand_ZeroOne() < 0.6f) {
+            this->flightParamsIdx = 1;
+        } else {
+            this->flightParamsIdx = 2;
+        }
+    } else {
+        this->flightParamsIdx = 0;
+    }
+
+    this->timer = Math_Rand_S16Offset(flightParams->minTime, flightParams->maxTime);
+}
+
+static f32 sTransformationEffectScale = 0.0f;
+static s16 sTransformationEffectAlpha = 0;
+
+void EnButte_ResetTransformationEffect(void) {
+    sTransformationEffectScale = 0.0f;
+    sTransformationEffectAlpha = 0;
+}
+
+void EnButte_UpdateTransformationEffect(void) {
+    sTransformationEffectScale += 0.003f;
+    sTransformationEffectAlpha += 4000;
+}
+
+void func_809CD144(EnButte* this, GlobalContext* globalCtx) {
+    static Vec3f D_809CE3C4 = { 0.0f, 0.0f, -3.0f };
+
+    Vec3f sp5C;
+    s32 alpha;
+    Vec3s camDir;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_choo.c", 295);
+
+    func_80093C14(globalCtx->state.gfxCtx);
+
+    alpha = Math_Sins(sTransformationEffectAlpha) * 250;
+    alpha = CLAMP(alpha, 0, 255);
+
+    func_8005A970(&camDir, ACTIVE_CAM);
+    Matrix_RotateY(camDir.y * (M_PI / 0x8000), MTXMODE_NEW);
+    Matrix_RotateX(camDir.x * (M_PI / 0x8000), MTXMODE_APPLY);
+    Matrix_RotateZ(camDir.z * (M_PI / 0x8000), MTXMODE_APPLY);
+    Matrix_MultVec3f(&D_809CE3C4, &sp5C);
+    func_800D1694(this->actor.posRot2.pos.x + sp5C.x, this->actor.posRot2.pos.y + sp5C.y,
+                  this->actor.posRot2.pos.z + sp5C.z, &camDir);
+    Matrix_Scale(sTransformationEffectScale, sTransformationEffectScale, sTransformationEffectScale, MTXMODE_APPLY);
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_choo.c", 317),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0x80, 0x80, 200, 200, 180, alpha);
+    gDPSetEnvColor(oGfxCtx->polyXlu.p++, 200, 200, 210, 255);
+    gSPDisplayList(oGfxCtx->polyXlu.p++, SEGMENTED_TO_VIRTUAL(&D_04010130));
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_choo.c", 326);
+}
+
+void EnButte_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static InitChainEntry sInitChain[] = {
+        ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
+        ICHAIN_F32(uncullZoneForward, 700, ICHAIN_CONTINUE),
```

Comment:
oh sorry i didnt mean to put it in the function, just above the init function is fine. we are doing that with these larger pieces of data for now